### PR TITLE
Initialize set with source in DFS

### PR DESCRIPTION
**Pull request by SandersLin**
### **Describe your change:**
checking if adjacent vertex is in explored set before pushing on to stack. The main difference between DFS and BFS is DFS uses a stack and BFS uses a FIFO queue. 

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by poyea** [comment]
I'll further fix this file.

File: graphs/dfs.py [hunk 1]
```diff
@@ -1,33 +1,34 @@
...
-    while stack:
-        v = (
-            stack.pop()
-        )  # one difference from BFS is to pop last element here instead of first one
 
-        if v in explored:
-            continue
-
-        explored.add(v)
+def dfs(graph: Dict, start: str) -> Set[int]:
```
Comment (cclauss) [inline graphs/dfs.py:18]:
```suggestion
def depth_first_search(graph: Dict, start: str) -> Set[int]:
```

File: graphs/dfs.py [hunk 2]
```diff
@@ -1,33 +1,34 @@
...
+    """Depth First Search on Graph
 
+       :param graph: directed graph in dictionary format
+       :param vertex: starting vectex as a string
+       :returns: the trace of the search
```
Comment (cclauss) [inline graphs/dfs.py:23]: Please add a doctest.

Comment (poyea) [reply graphs/dfs.py:23]: I'm not sure how to write doctest for this one. The function returns different outputs on different runs.

Comment (cclauss) [reply graphs/dfs.py:23]: Could we do:
```
>>> all(x in output for x in graph)
True
# and/or
>>> all(x in output for x in graph.values())
True
```

Comment (cclauss) [reply graphs/dfs.py:23]: If not then perhaps add a comment explaining that doctests would be difficult.

Comment (poyea) [reply graphs/dfs.py:23]: That works!

**Pull request merged**